### ISSUE TITLE
Referencing missing column

### ISSUE THREAD

**Issue by ohrrkan** [supabase/supabase-js#930]
@supabase/supabase-js": "^2.39.0" using Supabase CLI to [generate the types]   "supabase": "^1.123.0"

Query the same referenced table multiple times work but give you a Typescript error as it expect an array.

to reproduce you can use the datasource (doc example)

```
create table
 users (id int8 primary key, name text);

 create table
   messages (
     sender_id int8 not null references users,
     receiver_id int8 not null references users,
     content text
   );

 insert into
   users (id, name)
 values
   (1, 'Kiran'),
   (2, 'Evan');

 insert into
   messages (sender_id, receiver_id, content)
 values
   (1, 2, '👋');
```

query : 

```
const { data, error } = await supabase
  .from('messages')
  .select(`
    content,
    sender_id(name),
    receiver_id(name)
  `)
```

now using data[0].sender_id.name  will work but will give you an typescript error.  "sender_id: SelectQueryError<"Referencing missing column `name`">[];

Bypass temporary  solution : 

  type patch = { content: text; sender_id: Tables<"users">; receiver_id: Tables<"users">}[];

```
const { data, error } = await supabase
  .from('messages')
  .select(`
    content,
    sender_id(name),
    receiver_id(name)
  `).returns<patch>()
```

**Comment by dvsmc** [comment]
Is there any PR which fixes those? I recently updated supabase-js version and generaetd new types and got similar issue, some database fields are not recognized anymore.

**Comment by nabilhayek** [comment]
Same problem for me even when switching to 2.39.3
![image](https://github.com/supabase/supabase-js/assets/70241957/7729b860-14f0-49c0-abb4-52b1ebc3c9c9)

**Comment by RiodeJaneiroo** [comment]
i got the same error

**Comment by AndrianMauricio** [comment]
Same here!

**Comment by yaberkane05** [comment]
same here...

**Comment by rnnc** [comment]
still an issue

**Comment by artilishes** [comment]
Same here. Weirdly, this works...
```
supabase
  .from('selection_assets')
  .select(
    `
    id,
    comment,
    order,
    created_at,
    asset: assets (
      id,
      name,
      is_listed
    )
  `,
  )
  .eq('selection_id', selectionId)
  .single();
```
and this doesn't...
```
supabase
  .from('selections')
  .select(
    `
    *,
      client: client_id(id)
    `,
  )
  .eq('id', selectionId)
  .single();
```

**Comment by theshadowagent** [comment]
still an issue

**Comment by Th1nhNg0** [comment]
Got the issue too

**Comment by MattieTK** [comment]
Same issue

**Comment by silvaMatheus** [comment]
Any solution?

**Comment by karladler** [comment]
I have the feeling it depends on in which direction or on which side the relation is defined. Still need to verify this.

**Comment by rafal-r** [comment]
@ohrrkan As mentioned in the documentation here:
https://supabase.com/docs/reference/javascript/select
(example for 'Query referenced tables through a join table')

We can infer types by doing this:
```
// To infer types, use the name of the table (in this case `users`) and
// the name of the foreign key constraint.
const { data, error } = await supabase
  .from('messages')
  .select(`
    content,
    from:users!messages_sender_id_fkey(name),
    to:users!messages_receiver_id_fkey(name)
  `)
```

**Comment by yokowasis** [comment]
this works for me

```
  const { data, error } = await db
    .from("posts")
    .select(
      `
      id,
      content,
      user:users(
        id,
        full_name,
        handler,
        avatar
      )
      `
    )
```

`user` is the column name in `posts`
`users` is the name of foreign table.

**Comment by avallete** [comment]
A fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.

**Comment by Gabriel-Volpini** [comment]
Same issue

**Comment by avallete** [comment]
@KouroshBhl @Gabriel-Volpini

We had to revert the fix in `2.46.1` as we had some unexpected breaking changes due to the type rework.
However, we did released a new version as a "release candidate", and we are currently collecting feedback about it: https://github.com/orgs/supabase/discussions/30324

Please give `2.46.2-rc.3` a try, and don't hesitate to report back any remaining error with the debugging informations mentioned in the discussion. In the meantime I reopen this issue and will be closing it when this release candidate land on the main release channel (which should happen in December).

**Comment by avallete** [comment]
Happy to announce that we just released [v2.47.8](https://github.com/supabase/supabase-js/releases/tag/v2.47.8) that include the typing inference rework which should fix this issue.
